datafusion/pruning/src/pruning_predicate.rs

```diff
@@ -1369,7 +1368,7 @@ impl PredicateRewriter {
         let mut required_columns = RequiredColumns::new();
         build_predicate_expression(
             expr,
-            schema,
+            &Arc::new(schema.clone()),
```

Comment:
Seems like in this case we've just moved the problem up a level. Should we edit `rewrite_predicate_to_statistics_predicate` to also accept an `&SchemaRef`? cc @xudong963